bundles/binding/org.openhab.binding.maxcube/src/main/java/org/openhab/binding/maxcube/internal/MaxCubeBinding.java

```diff
@@ -186,10 +199,14 @@ public synchronized void execute() {
 					cont = false;
 					continue;
 				}
-
-				Message message;
+				
+				Message message = null;
 				try {
-					message = processRawMessage(raw);
+					if (this.messageProcessor.addReceivedLine(raw)) {
```

Comment:
Is it not more elegant to add a messageProcessor.IsMessageAvailable() test instead?